src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -484,12 +491,11 @@ private BulkResponse execute() throws Exception {
             if (reporter != null) {
               for (R record : batch) {
                 SinkRecord original = recordsToReportOnError.get(record);
-                if (original != null) {
+                BulkResultItem result = bulkRsp.failedRecords.get(record);
+                String error = result != null ? result.error : null;
+                if (original != null && error != null) {
```

Comment:
assuming that in the records reporter you have added all the batch and only a few records have errored out, let's first check error
```suggestion
                if (error != null && original != null) {
```